pandas/tests/io/parser/test_python_parser_only.py

```diff
@@ -329,3 +332,73 @@ def readline(self):
             return self.data
 
     parser.read_csv(NoNextBuffer("a\n1"))
+
```

Comment:
Could you add a test for the following cases
- index_col inferred
- usecols (I think usecols takes precedence, should maybe add a note in the userguide)

Reply:
Isn't `index_col` always inferred since it's set to `None` in these tests?

Also in the Python parser, appears that `on_bad_lines` is never considered if `usecols is not None` because of this line

https://github.com/pandas-dev/pandas/blob/8ae1b4c2880e57ece25483fa7970e7090e474739/pandas/io/parsers/python_parser.py#L979

Reply:
Ah good, then don't need tests.

Yeah, but most of the time it is None (meaning index_col). This is only not None if something like
data = """a,b
1,2,3
4,5,6
"""
is passed as input

Reply:
Ah okay added a tests with 

data = """a,b
1,2,3
4,5,6
"""

as an input

Reply:
thx